be/src/service/service_be/lake_service.cpp

```diff
@@ -694,12 +696,21 @@ void LakeServiceImpl::get_tablet_stats(::google::protobuf::RpcController* contro
                     for (const auto& [_, file] : (*tablet_metadata)->delvec_meta().version_to_file()) {
                         data_size += file.size();
                     }
+                    auto res =
+                            _tablet_mgr->collect_tablet_storage_size(tablet_id, version, vacuum_version, min_version);
```

Comment:
this will be triggered by the get_tablet_stats call, which will be collected and reported by BE service repeatedly. With this change, every stats will touch all the tablet metas no matter it is the latest version or not, it can cause a few problems:

1. additional remote GET ops to the versioned tablet meta.
2. fill in the metadata cache with historic version of tablet meta that may not be used at all anywhere else by here. makes the metadata cache almost useless.

Reply:
> this will be triggered by the get_tablet_stats call, which will be collected and reported by BE service repeatedly. With this change, every stats will touch all the tablet metas no matter it is the latest version or not, it can cause a few problems:
> 
> 1. additional remote GET ops to the versioned tablet meta.
> 2. fill in the metadata cache with historic version of tablet meta that may not be used at all anywhere else by here. makes the metadata cache almost useless.

Agree with you, i will try to find a better solution.